test/test_max.js

```diff
@@ -0,0 +1,51 @@
+require('../dist/array_extensions');
+var assert = require('chai').assert;
+var sinon = require('sinon');
+var jsonfile  =require('jsonfile');
+
+var file = './fixtures/data.json';
+var fixtures = jsonfile.readFileSync(file);
+
+
+describe('Array #max', function () {
+	it('Should return empty', function () {
+		//Execute
+		let result = fixtures.arrayEmpty.max();
+
+		//Compare
+		assert.equal(result, null);
+	});
+
+	it('Should use comparator to find max element', function() {
+		//Setup
+		let expected = { "name": "pablo", "age": 20 };
+		let callback = (a, b) => a.age - b.age;
```

Comment:
We can make it more semantically coherent, in the sense of matching implementation + documentation + tests. I mean, if the `max` function is waiting for a *comparator* we musts name it that way, it will improve readability and reduce learning curve for the user of the API and that reflects on more adoption.

```js
let comparator = (a, b) => a.age - b.age;
//Execute
let result = fixtures.peopleRandom.max(comparator);
```

Some users finds tests as the more useful tutorials.